### PR TITLE
Support serve results to prometheus before ready

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: pkg/collector/collector.go [hunk 1]
```diff
@@ -1584,6 +1587,11 @@ func (c *Collector) registerWithSamplers(ctx context.Context) {
 
 				readyPct := (nodesWithResults.Len() * 100 / nodesWithSamplers.Len())
 				if nodesWithSamplers.Len() > 0 && readyPct > c.UtilizationServer.MinResultPctBeforeReady {
+					cyclesCount++
+					if cyclesCount > c.UtilizationServer.WaitSamplerRegistrationsBeforeServe {
+						c.UtilizationServer.ServeResults.Store(true)
+					}
```
Comment (ichekrygin) [inline pkg/collector/collector.go:1593]: Possibly, could be:
```suggestion
					c.UtilizationServer.ServeResults.Store(cyclesCount > c.UtilizationServer.WaitSamplerRegistrationsBeforeServe)

```

File: pkg/collector/collector.go [hunk 2]
```diff
@@ -1584,6 +1587,11 @@ func (c *Collector) registerWithSamplers(ctx context.Context) {
...
+
```
Comment (ichekrygin) [inline pkg/collector/collector.go:1594]: Not part of this PR.
To avoid "repetition." you can consider: 
```suggestion
log := log.WithValues("running-minutes", time.Since(c.startTime).Minutes(),
						"nodes-with-results-count", nodesWithResults.Len(),
						"nodes-with-samplers-count", nodesWithSamplers.Len(),
						"nodes-with-running-samplers-count", nodesWithRunningSamplers.Len(),
						"nodes-missing-results", nodesMissingResults.List(),
						"ready-pct", readyPct,
						"min-ready-pct", c.UtilizationServer.MinResultPctBeforeReady,
						"remaining-cycles", waitReadyCyclesCount)
```
with following: 
- `log.Info("collector ready")`
- `log.Info("collector has required sampler results, waiting on WaitSamplerRegistrationsBeforeReady to be ready")`
- `log.Info("collector not-ready")`

Note: `log := log.WithValues` will only be applicable in this closure.

**Review by ichekrygin — APPROVED** [review]
LGTM, with a question and a couple of nits/suggestions (nothing that should prevent the merge).

File: pkg/collector/collector.go [hunk 3]
```diff
@@ -1584,6 +1587,11 @@ func (c *Collector) registerWithSamplers(ctx context.Context) {
 
 				readyPct := (nodesWithResults.Len() * 100 / nodesWithSamplers.Len())
 				if nodesWithSamplers.Len() > 0 && readyPct > c.UtilizationServer.MinResultPctBeforeReady {
+					cyclesCount++
+					if cyclesCount > c.UtilizationServer.WaitSamplerRegistrationsBeforeServe {
+						c.UtilizationServer.ServeResults.Store(true)
```
[the remark this answers is not in the document]
Comment (pwittrock) [reply pkg/collector/collector.go:1592]: right, I am not sure when we would want to do this.  This is mostly to control the rollout and deletion of pods so that we are fully initialized with node data before sending it to prometheus.
